### DISCUSSION TITLE
[Q] How to create a multiple-build-systems package based on different versions?

### DISCUSSION THREAD

**Discussion by hyoklee** [spack/spack#35589] [Q&A] [answered]
Hi,
  
  I'm working on DPDK package.
It used Makefile before but it now uses Meson.

  In this case, how can I create a package that supports both?
I wish I could invoke meson instead of make if a user specifies a later version.

  Is there any good example package that invokes different build system based on version?

**Comment by haampie** [comment, marked as answer]
In Spack 0.19 and above you can do this. See the docs here https://spack.readthedocs.io/en/latest/packaging_guide.html#multiple-build-systems. You can also grep the existing packages for `build_system` to see examples.

**Reply by hyoklee** [reply]
Thanks @haampie about the "grep" tip and immediate help! 
GDAL package example looks really good and fits my need.
